### PR TITLE
SCB-1383 support toolkit maven plugin commands that generating code

### PR DESCRIPTION
**Pull request by MabinGo**
Signed-off-by: MabinGo <bin.ma@huawei.com>

### CODE REVIEW

**Comment by MabinGo** [comment]
@WillemJiang pls help to check,thks:  )

File: toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateMojo.java [hunk 1]
```diff
@@ -76,14 +76,14 @@ public void execute() {
         GenerateUtil.generateContract(project, contractOutput, contractFileType, "default");
         contractLocation = contractOutput;
         if (Objects.requireNonNull(new File(contractOutput).listFiles()).length == 0) {
-          LOGGER.info("No contract in the code");
+          LOGGER.info("no contract in the code");
```
Comment (WillemJiang) [inline toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateMojo.java]: 日志需要首字母需要大写

File: toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateMojo.java [hunk 2]
```diff
@@ -92,7 +92,20 @@ public void execute() {
 
         break;
       default:
-        throw new RuntimeException("source type " + sourceType + " is not supported");
+        throw new RuntimeException("not support source type " + sourceType);
```
Comment (WillemJiang) [inline toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateMojo.java]: 异常信息首字母也是需要大写。

File: toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateMojo.java [hunk 3]
```diff
@@ -92,7 +92,20 @@ public void execute() {
...
+    }
+
+    //generate microservice project
+    if (service == null) {
+      LOGGER.info("no service configuration and do not generate code");
```
Comment (WillemJiang) [inline toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateMojo.java]: ”Cannot generate code without service configuration!"

File: toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateUtil.java [hunk 4]
```diff
@@ -57,16 +58,17 @@ public static void generateContract(MavenProject project, String contractOutput,
...
 
-  public static void generateDocument(String contractLocation, String documentOutput, String type)
-      throws IOException {
+  public static void generateDocument(String contractLocation, String documentOutput, String type) throws IOException {
 
     // TODO: support users to add other getGenerator type soon
     DocGenerator docGenerator = GeneratorFactory.getGenerator(DocGenerator.class, type);
     if (docGenerator == null) {
-      throw new RuntimeException("DocGenerator's implementation is not found");
+      throw new RuntimeException("not found document generator's implementation");
```
Comment (WillemJiang) [inline toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateUtil.java]: "Cannot xxx"

File: toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateUtil.java [hunk 5]
```diff
@@ -81,10 +83,55 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
             .substring(0, file.toFile().getName().indexOf(".")));
 
         docGenerator.configure(docGeneratorConfig);
-        docGenerator.generate();
+        if (!docGenerator.generate()) {
+          throw new RuntimeException("failed to generate document by generator " + type);
```
Comment (WillemJiang) [inline toolkit-maven-plugin/src/main/java/org/apache/servicecomb/toolkit/plugin/GenerateUtil.java]: Can we add more specific information here? 
You can just throw the exception with detail information inside of generate method of docGenerator.

**Comment by MabinGo** [comment]
@WillemJiang all code review suggestions are fixed,pls check again,thks.

File: contractgen/src/test/java/org/apache/servicecomb/toolkit/contractgen/DefaultContractsGeneratorTest.java [hunk 6]
```diff
@@ -85,7 +86,11 @@ public void testGenerate()
 
     DefaultContractsGenerator defaultContractsGenerator = new DefaultContractsGenerator();
     defaultContractsGenerator.configure(config);
-    assertTrue(defaultContractsGenerator.generate());
+    try {
+      defaultContractsGenerator.generate();
+    } catch (RuntimeException e) {
+      fail();
```
Comment (WillemJiang) [inline contractgen/src/test/java/org/apache/servicecomb/toolkit/contractgen/DefaultContractsGeneratorTest.java]: It's better to add some information which case failure in the fail method. 
It's could save us lot of time to find out the CI build error.

**Pull request merged**